### PR TITLE
[ROCm] add fpus for rocm and enable xla_compile_lib_test

### PR DESCRIPTION
**Pull request by i-chaochen**
1. we add `fpus_per_core` for cost model and fusion analysis.
2. enable ROCm for new xla_compile_lib_test https://github.com/openxla/xla/commit/a129c9bfe093743a93a9c2986d187a066940c21d#diff-19845b853b7cf43ac606aca22262c15fdf2d56e5eb196a98f8d982e947123f1d

please check @xla-rotation thanks in advance!

### CODE REVIEW

**Review by jvstokes — APPROVED** [review]
Thank you!

**Comment by i-chaochen** [comment]
@xla-rotation wonder when can we merge this PR?  Thanks!

**Comment by ddunl** [comment]
Unfortunately it is pending special approval right now (but should be no problems after it gets that approval)

**Pull request closed without merging**
